File: Sem3/Advanced Methods of Programming/MyInterpreter - CSharp Console Application/MyInterpreter CSharp/domain/adt/MyList.cs
using System.Collections.Generic;

namespace MyInterpreter_CSharp.domain.adt
{
    public class MyList<TElement> : IMyList<TElement>
    {
        private Queue<TElement> _list;

        public MyList()
        {
            _list = new Queue<TElement>();
        }

        public override void Add(TElement element)
        {
            _list.Enqueue(element);
        }

        public override TElement GetFirstElement()
        {
            if (_list.Peek() != null)
                return _list.Peek();
            throw new AdtException("Peek on empty list.");
        }

        public override string ToString()
        {
            string str = "";
            foreach (var element in _list)
            {
                str += element + "\n";
            }

            return str;
        }
    }
}

File: Sem3/Advanced Methods of Programming/MyInterpreter - JavaFX - Final Project/project/src/domain/adt/MyList.java
package domain.adt;


import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;


public class MyList<TElement> implements MyListInterface<TElement>
{
    private Queue<TElement> list;


    public MyList()
    {
        list = new LinkedList<TElement>();
    }


    @Override
    public void add(TElement element)
    {
        list.add(element);
    }


    @Override
    public TElement pop() throws AdtException
    {
        if (list.isEmpty())
            throw new AdtException("Pop from empty list.");
        return list.poll();
    }


    @Override
    public TElement getFirstElement() throws AdtException
    {
        if (list.peek() != null)
            return list.peek();
        throw new AdtException("Peek on empty list.");
    }


    @Override
    public String toString()
    {
        String str = "";
        for (TElement element : list)
        {
            str += element.toString() + "\n";
        }
        return str;
    }


    @Override
    public List<TElement> getAll()
    {
        return new ArrayList<>(list);
    }
}